File: Requests/QingGuo/MultiAPI/sunliying_test.py
import unittest
from core.Sendhttp import SendHttp
from core import Common


class qgSenceTest(unittest.TestCase):
    def setUp(self):
        self.LoginUrl = "/common/fgadmin/login"
        self.AddressListUrl = "/fgadmin/address/list"
        self.SkuListUrl = "/common/skuList"
        self.TransportFeeUrl = "/common/getTransportFee"
        self.NewAddressUrl = "/fgadmin/address/new"
        self.SubmitUrl = "/fgadmin/orders/submit"
        self.DeleteAddress = "/fgadmin/address/delete"

    # 登录成功 - -获取skuList - -获取list - -计算运费 - -新建地址 - -提交订单 - -删除地址
    def test_scene1(self):
        # 登录
        user = {"phoneArea": "86", "phoneNumber": "200000000044", "password": "netease123"}
        resultLogin = SendHttp().run_http(self.LoginUrl, "POST", user)
        self.assertEqual(resultLogin['code'], 200)
        self.assertEqual(resultLogin['message'], 'success')

        # 获取skuList,goodsId=1
        goodsId = 1
        resultSkuList = SendHttp().run_http(self.SkuListUrl + "?" + str(goodsId), "GET")
        self.assertEqual(resultSkuList['code'], 200)
        self.assertEqual(resultSkuList['message'], 'success')

        # 实现获取list
        address_result = SendHttp().sent_get_bycookies(self.AddressListUrl, Common.getcookies(user))
        pro = address_result['result']['list'][0]['province']
        city = address_result['result']['list'][0]['city']
        area = address_result['result']['list'][0]['area']
        self.assertEqual(address_result['code'], 200)
        self.assertEqual(address_result['message'], 'success')

        # 获取运费
        para = {"id": 1, "addressDetail": "河北省_石家庄市_裕华区"}
        resultFee = SendHttp().run_http(self.TransportFeeUrl, "GET", para)
        self.assertEqual(resultFee['code'], 200)
        self.assertEqual(resultFee['message'], 'success')

        # 新建地址
        para = {"receiverName": "张三",
                "cellPhone": "12345678901",
                'addressDetail': "浙江大学",
                "province": "浙江省",
                "city": "杭州市",
                "area": "滨江区"}
        resultNewAddress = SendHttp().sent_post_bycookies(self.NewAddressUrl, Common.getcookies(user), para)
        self.assertEqual(resultNewAddress['code'], 200)
        self.assertEqual(resultNewAddress['message'], 'success')

        # 提交订单
        submit_data = {"skuIds": "2,3",
                       "skuNumbers": "1,1",
                       "stockIds": "74966312,74966313",
                       "receiverName": "张三",
                       "cellPhone": "12615813537",
                       "addressDetail": "1 栋 3 单元",
                       "province": pro,
                       "city": city,
                       "area": area,
                       "voiceStatus": 0,
                       "needInvoice": 0,
                       "invoiceHead": "",
                       "transportFee": 0,
                       "logisticsCompanyId": 1,
                       "accessSource": "noSource",
                       "accessDevice": 0}
        resultSubmit = SendHttp().sent_post_bycookies(self.SubmitUrl, Common.getcookies(user), submit_data)
        self.assertEqual(resultSubmit['code'], 200)
        self.assertEqual(resultSubmit['message'], 'success')

    # 登录成功 - -新建订单 - -删除订单
    def test_scene2(self):
        # 登录成功
        user = {"phoneArea": "86", "phoneNumber": "200000000044", "password": "netease123"}
        result = SendHttp().run_http(self.LoginUrl, "POST", user)
        self.assertEqual(result['code'], 200)
        self.assertEqual(result['message'], 'success')
        # 新建订单
        para = {"receiverName": "张三",
                "cellPhone": "12345678901",
                'addressDetail': "浙江大学",
                "province": "浙江省",
                "city": "杭州市",
                "area": "滨江区"}
        SendHttp().sent_post_bycookies(self.NewAddressUrl, Common.getcookies(user), para)
        self.assertEqual(result['code'], 200)
        self.assertEqual(result['message'], 'success')
        # 删除订单
        address_result = SendHttp().sent_get_bycookies(self.AddressListUrl, Common.getcookies(user))
        aid = address_result['result']['list'][0]['id']
        para = {"id": aid}
        SendHttp().sent_post_bycookies(self.DeleteAddress, Common.getcookies(user), para)
        self.assertEqual(address_result['code'], 200)
        self.assertEqual(address_result['message'], 'success')

    # 登录成功 - -新建订单 - -提交订单
    def test_scene3(self):
        # 登录成功
        user = {"phoneArea": "86", "phoneNumber": "200000000044", "password": "netease123"}
        result = SendHttp().run_http(self.LoginUrl, "POST", user)
        self.assertEqual(result['code'], 200)
        self.assertEqual(result['message'], 'success')
        # 新建订单
        para = {"receiverName": "张三",
                "cellPhone": "12345678901",
                'addressDetail': "浙江大学",
                "province": "浙江省",
                "city": "杭州市",
                "area": "滨江区"}
        SendHttp().sent_post_bycookies(self.NewAddressUrl, Common.getcookies(user), para)
        self.assertEqual(result['code'], 200)
        self.assertEqual(result['message'], 'success')
        # 提交订单
        submit_data = {"skuIds": "2,3",
                       "skuNumbers": "1,1",
                       "stockIds": "74966312,74966313",
                       "receiverName": "张三",
                       "cellPhone": "12615813537",
                       "addressDetail": "1 栋 3 单元",
                       "province": "1",
                       "city": "1",
                       "area": "1",
                       "voiceStatus": 0,
                       "needInvoice": 0,
                       "invoiceHead": "",
                       "transportFee": 0,
                       "logisticsCompanyId": 1,
                       "accessSource": "noSource",
                       "accessDevice": 0}
        resultSubmit = SendHttp().sent_post_bycookies(self.SubmitUrl, Common.getcookies(user), submit_data)
        self.assertEqual(resultSubmit['code'], 200)
        self.assertEqual(resultSubmit['message'], 'success')

    # 登录失败 - -查询收货地址失败
    def test_scene4(self):
        # 登录失败
        user = {"phoneArea": "86", "phoneNumber": "200000000044", "password": "netease1234"}
        result = SendHttp().run_http(self.LoginUrl, "POST", user)
        self.assertEqual(result['code'], 400)
        self.assertEqual(result['message'], '用户名或者密码错误')
        # 查询收货地址list失败
        address_result = SendHttp().sent_get_bycookies(self.AddressListUrl, Common.getcookies(user))
        self.assertEqual(address_result['code'], 403)
        self.assertEqual(address_result['message'], '请重新登录')

    # 登录成功 - -查询地址 - -计算运费 - -新建订单 - -提交订单 - -删除订单
    def test_scene5(self):
        # 登录成功
        user = {"phoneArea": "86", "phoneNumber": "200000000044", "password": "netease123"}
        result = SendHttp().run_http(self.LoginUrl, "POST", user)
        self.assertEqual(result['code'], 200)
        self.assertEqual(result['message'], 'success')
        # 查询地址
        address_result = SendHttp().sent_get_bycookies(self.AddressListUrl, Common.getcookies(user))
        pro = address_result['result']['list'][0]['province']
        city = address_result['result']['list'][0]['city']
        area = address_result['result']['list'][0]['area']
        self.assertEqual(address_result['code'], 200)
        self.assertEqual(address_result['message'], 'success')
        # 计算运费
        para = {"id": 1, "addressDetail": pro + "_" + city + "_" + area}
        resultFee = SendHttp().run_http(self.TransportFeeUrl, "GET", para)
        self.assertEqual(resultFee['code'], 200)
        self.assertEqual(resultFee['message'], 'success')
        # 新建订单
        para = {"receiverName": "张三",
                "cellPhone": "12345678901",
                'addressDetail': "浙江大学",
                "province": "浙江省",
                "city": "杭州市",
                "area": "滨江区"}
        SendHttp().sent_post_bycookies(self.NewAddressUrl, Common.getcookies(user), para)
        self.assertEqual(result['code'], 200)
        self.assertEqual(result['message'], 'success')
        # 提交订单
        submit_data = {"skuIds": "2,3",
                       "skuNumbers": "1,1",
                       "stockIds": "74966312,74966313",
                       "receiverName": "张三",
                       "cellPhone": "12615813537",
                       "addressDetail": "1 栋 3 单元",
                       "province": pro,
                       "city": city,
                       "area": area,
                       "voiceStatus": 0,
                       "needInvoice": 0,
                       "invoiceHead": "",
                       "transportFee": 0,
                       "logisticsCompanyId": 1,
                       "accessSource": "noSource",
                       "accessDevice": 0}
        resultSubmit = SendHttp().sent_post_bycookies(self.SubmitUrl, Common.getcookies(user), submit_data)
        self.assertEqual(resultSubmit['code'], 200)
        self.assertEqual(resultSubmit['message'], 'success')
        # 删除订单
        address_result = SendHttp().sent_get_bycookies(self.AddressListUrl, Common.getcookies(user))
        aid = address_result['result']['list'][0]['id']
        para = {"id": aid}
        SendHttp().sent_post_bycookies(self.DeleteAddress, Common.getcookies(user), para)
        self.assertEqual(address_result['code'], 200)
        self.assertEqual(address_result['message'], 'success')


if __name__ == '__main__':
    unittest.main()

File: QingGuoAPITest/src/com/edu/qgMultiAPI/SunLiyingTest.java
package com.edu.qgMultiAPI;

import java.io.IOException;
import java.util.HashMap;
import java.util.Map;

import org.apache.http.client.CookieStore;
import org.testng.annotations.Test;

import com.edu.common.Common;
import com.edu.core.BaseTest;
import com.edu.core.HttpDriver;
import com.edu.utils.Checker;
import net.sf.json.JSONObject;

/*
 * @author=孙立莹
 */
public class SunLiyingTest extends BaseTest {
	String result = null;
	String name = "200000000044";
	String pwd = "netease123";
	Common common = new Common();
	Checker check;

//	登录
	public void LoginSuccess(String name, String pwd) throws Exception {
		String loginUrl = baseUrl + common.loginUrl;
		JSONObject js = new JSONObject();
		js.element("password", pwd);
		js.element("phoneArea", 86);
		js.element("phoneNumber", name);
		result = HttpDriver.doPost(loginUrl, js);
		System.out.println(result);
		check = new Checker(result);
		check.verifyXpath("message", "success");
	}

//	登录失败
	public void LoginFail(String name, String pwd) throws Exception {
		String loginUrl = baseUrl + common.loginUrl;
		JSONObject js = new JSONObject();
		js.element("password", pwd);
		js.element("phoneArea", 86);
		js.element("phoneNumber", name);
		result = HttpDriver.doPost(loginUrl, js);
		System.out.println(result);
		check = new Checker(result);
		check.verifyXpath("message", "用户名或者密码错误");
	}

//	获取skuList
	public void SkuList() throws Exception {
		int goodsId = 1;
		result = HttpDriver.doGet(baseUrl + common.SkuListUrl + "?goodsId" + Integer.toString(goodsId));
		System.out.println(result);
		check = new Checker(result);
		check.verifyXpath("message", "success");
	}

//	获取list成功
	public void ListSuccess() throws Exception {
		CookieStore cook = Common.getLoginCookie(name, pwd);
		result = HttpDriver.doGet(baseUrl + common.AddressListUrl, cook);
		System.out.println(result);
		check = new Checker(result);
		check.verifyXpath("message", "success");
	}

//	获取list失败
	public void ListFail(String name, String pwd) throws Exception {
		CookieStore cook = Common.getLoginCookie(name, pwd);
		result = HttpDriver.doGet(baseUrl + common.AddressListUrl, cook);
		System.out.println(result);
		check = new Checker(result);
		check.verifyXpath("code", "403");
	}

//	计算运费
	public void getFee() throws Exception {
		String feeUrl = baseUrl + common.TransportFeeUrl;
		Map<String, Object> map = new HashMap<String, Object>();
		map.put("id", 1);
		map.put("addressDetial", "河北省_石家庄市_裕华区");
		result = HttpDriver.doGet(feeUrl, map);
		System.out.println(result);
		check = new Checker(result);
		check.verifyXpath("message", "success");
	}

//	新建订单
	public void newAddress() throws IOException, Exception {
		String addAddressUrl = baseUrl + common.NewAddressUrl;
		CookieStore cook = Common.getLoginCookie(name, pwd);
		JSONObject json = new JSONObject();
		json.element("id", "");
		json.element("receiverName", "张三");
		json.element("addressDetail", "河北省_石家庄市_裕华区");
		json.element("cellPhone", "123456789");
		json.element("province", "河北省");
		json.element("city", "石家庄市");
		json.element("area", "裕华区");
		result = HttpDriver.doPost(addAddressUrl, json, cook);
		System.out.println(result);
		check = new Checker(result);
		check.verifyXpath("message", "success");
	}

//	提交订单
	public void submit() throws IOException, Exception {
		String submitUrl = baseUrl + common.SubmitUrl;
		CookieStore cook = Common.getLoginCookie(name, pwd);
		JSONObject json = new JSONObject();
		json.element("skuIds", "2,3");
		json.element("skuNumbers", "1,1");
		json.element("stockIds", "74966312,74966313");
		json.element("receiverName", "张三");
		json.element("cellPhone", "1234566");
		json.element("addressDetail", "1栋3单元");
		json.element("province", "浙江省");
		json.element("city", "杭州市");
		json.element("area", "滨江区");
		json.element("voiceStatus", "0");
		json.element("needInvoice", "0");
		json.element("invoiceHead", "1");
		json.element("transportFee", "0");
		json.element("logisticsCompanyId", "1");
		json.element("accessSource", "noSource");
		json.element("accessDevice", "0");
		result = HttpDriver.doPost(submitUrl, json, cook);
		System.out.println(result);
		check = new Checker(result);
		check.verifyXpath("message", "success");
	}

//	删除订单
	public void delete() throws IOException, Exception {
		String deleteUrl = baseUrl + common.DeleteAddressUrl;
		CookieStore cook = Common.getLoginCookie(name, pwd);
		JSONObject json = new JSONObject();
		json.element("id", 82319421);
		result = HttpDriver.doPost(deleteUrl, json, cook);
	}

	@Test(description = "登录成功--获取skuList成功--获取list成功--计算运费--新建订单--提交订单--删除订单")
	public void test_scence1() throws Exception {
		String name = "200000000044";
		String pwd = "netease123";
		this.LoginSuccess(name, pwd);
		this.SkuList();
		this.ListSuccess();
		this.getFee();
		this.newAddress();
		this.submit();
		this.delete();
	}

	@Test(description = "登录成功--新建地址--删除订单")
	public void test_scence2() throws Exception {
		String name = "200000000044";
		String pwd = "netease123";
		this.LoginSuccess(name, pwd);
		this.newAddress();
		this.delete();
	}

	@Test(description = "登录成功--新建订单--提交订单")
	public void test_scence3() throws Exception {
		String name = "200000000044";
		String pwd = "netease123";
		this.LoginSuccess(name, pwd);
		this.newAddress();
		this.submit();
	}

	@Test(description = "登录失败--查询收货地址失败")
	public void test_scence4() throws Exception {
		String name = "200000000044011";
		String pwd = "netease123";
		this.LoginFail(name, pwd);
		this.ListFail(name, pwd);
	}

	@Test(description = "登录成功--查询地址--计算运费--新建订单--提交订单--删除订单")
	public void test_scence5() throws Exception {
		String name = "200000000044";
		String pwd = "netease123";
		this.LoginSuccess(name, pwd);
		this.ListSuccess();
		this.getFee();
		this.newAddress();
		this.delete();
	}
}
